presto-spark-base/src/main/java/com/facebook/presto/spark/PrestoSparkConfig.java

```diff
@@ -15,10 +15,29 @@
 
 import com.facebook.airlift.configuration.Config;
 import com.facebook.airlift.configuration.ConfigDescription;
+import io.airlift.units.DataSize;
+
+import static io.airlift.units.DataSize.Unit.GIGABYTE;
 
 public class PrestoSparkConfig
 {
+    private boolean autoTuneSparkPartitionCount = true;
     private int initialSparkPartitionCount = 16;
+    private int minSplitsPerSparkPartition = Runtime.getRuntime().availableProcessors();
```

Comment:
Curious: Will this be the total number of physical cores on the Spark workers (instead of Spark container CPUs? ). (E.g. the worker has 16 cores, but the Spark container only has 4 CPUs, and this will set `minSplitsPerSparkPartition` to 16? )

Reply:
@arhimondr Do you know the answer?  I can run some test to verify.

Reply:
Yeah, I think this default doesn't make a lot of sense.

I think it is even better to do not set any defaults and keep the autotune disabled. Then it can be manually enabled and the correct settings will be set.